include/can.h

```diff
@@ -0,0 +1,380 @@
+							   u32_t bitrate);
+
+/**
+ * @brief Perform data transfer to CAN bus.
+ *
+ * This routine provides a generic interface to perform data transfer
+ * to the can bus. Use can_write() for simple write.
+ * *
+ * @param dev          Pointer to the device structure for the driver instance.
+ * @param msg          Message to transfer.
+ * @param timeout      Waiting for empty tx mailbox timeout in ms or K_FOREVER.
+ * @param callback_isr Is called when message was sent or a transmission error
+ *                     occurred. If null, this function is blocking until
+ *                     message is sent.
+ *
+ * @retval 0 If successful.
+ * @retval CAN_TX_* on failure.
+ */
+typedef int (*can_send_t)(struct device *dev, struct can_msg *msg,
+						  s32_t timeout, can_tx_callback_t callback_isr);
+
+
+/**
+ * @brief Attach a message queue to a single or group of identifiers.
+ *
+ * This routine attaches a message queue to identifiers specified by
+ * a filter. Whenever the filter matches, the message is pushed to the queue
+ * If a message passes more than one filter the priotity of the match
+ * is hardware dependent.
+ * A message queue can be attached to more than one filter.
+ * The message queue must me initialized before.
+ * *
+ * @param dev    Pointer to the device structure for the driver instance.
+ * @param msgq   Pointer to the already initialized message queue.
+ * @param filter Pointer to a can_filter structure defining the id filtering.
+ *
+ * @retval filter id on success.
+ * @retval CAN_NO_FREE_FILTER if there is no filter left.
+ */
+typedef int (*can_attach_msgq_t)(struct device *dev, struct k_msgq *msg_q,
+								 struct can_filter *filter);
+
+/**
+ * @brief Attach a isr callback function to a single or group of identifiers.
+ *
+ * This routine attaches a isr callback to identifiers specified by
+ * a filter. Whenever the filter matches, the callback function is called
+ * with isr context.
+ * If a message passes more than one filter the priotity of the match
+ * is hardware dependent.
+ * A callback function can be attached to more than one filter.
+ * *
+ * @param dev    Pointer to the device structure for the driver instance.
+ * @param isr    Callback functionpointer.
+ * @param filter Pointer to a can_filter structure defining the id filtering.
+ *
+ * @retval filter id on success.
+ * @retval CAN_NO_FREE_FILTER if there is no filter left.
+ */
+typedef int (*can_attach_isr_t)(struct device *dev, can_rx_callback_t isr,
+				struct can_filter *filter);
+
+/**
+ * @brief Detach a isr or message queue from the identifier filtering.
+ *
+ * This routine detaches a isr callback  or message queue from the identifier
+ * filtering.
+ * *
+ * @param dev       Pointer to the device structure for the driver instance.
+ * @param filter_id filter id returned by can_attach_isr or can_attach_msgq.
+ *
+ * @retval none
+ */
+typedef void (*can_detach_t)(struct device *dev, int filter_nr);
```

Comment:
s/filter_nr/filter_id ?

Reply:
filter_id is the right one. I typed this by mistake because in my stm32 implementation its the same.